src/USPS/USPSBase.php

```
@@ -208,9 +228,117 @@ protected function doRequest($ch = null)
+            $this->setErrorMessage('Failed to get access token: ' . $e->getMessage());
+            return false;
+        }
+    }
+
+  /**
+   * Make v3 API request
+   * @param string $endpoint
+   * @param array $params
+   * @return string
+   */
+    protected function doRequestV3($endpoint, $params = [])
+    {
+        $accessToken = $this->getAccessToken();
+        if (!$accessToken) {
+            return '';
+        }
+
+        try {
+            $client = new Client(['timeout' => 60]);
+            $url = $this->getEndpoint() . $endpoint;
+
+            $response = $client->get($url, [
+                'query' => $params,
+                'headers' => [
+                    'Authorization' => 'Bearer ' . $accessToken,
+                    'Accept' => 'application/json',
+                ]
+            ]);
+
+            $body = $response->getBody()->getContents();
+            $this->setResponse($body);
+            $this->setHeaders([
+                'http_code' => $response->getStatusCode(),
+                'content_type' => $response->getHeaderLine('Content-Type')
+            ]);
+
+            $jsonData = json_decode($body, true);
+            if ($jsonData !== null) {
+                $this->setArrayResponse($jsonData);
+            } else {
+                $this->setErrorMessage('Failed to parse JSON: ' . json_last_error_msg());
+            }
+
+            if ($this->isError()) {
+                $arrayResponse = $this->getArrayResponse();
+                if (isset($arrayResponse['error'])) {
+                    $this->setErrorCode($arrayResponse['error']['code'] ?? 0);
+                    $this->setErrorMessage($arrayResponse['error']['message'] ?? 'Unknown error');
+                }
+            }
+
+            return $this->getResponse();
+        } catch (GuzzleException $e) {
+            $this->setErrorCode($e->getCode());
+            $this->setErrorMessage('API request failed: ' . $e->getMessage());
+            return '';
```

If possible, can we stop returning `false` or other things to indicate failure, and instead let the error propagate to the calling code?